spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -290,6 +311,9 @@ protected DatagramSocket getTheSocket() {
 	}
 
 	protected synchronized DatagramSocket getSocket() throws IOException {
+		if (this.socketExpression != null) {
+			return this.socketExpression.getValue(this.evaluationContext, DatagramSocket.class);
```

Comment:
Yes; you are correct; my suggested implementation only supports this narrow use case.